src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -163,28 +168,37 @@ export class ElasticsearchService implements CoreService<InternalElasticsearchSe
     };
 
     const esNodesCompatibility$ = pollEsNodesVersion({
-      callWithInternalUser: adminClient.callAsInternalUser,
+      callWithInternalUser: this.adminClient.callAsInternalUser,
       log: this.log,
       ignoreVersionMismatch: config.ignoreVersionMismatch,
       esVersionCheckInterval: config.healthCheckDelay.asMilliseconds(),
       kibanaVersion: this.kibanaVersion,
     }).pipe(takeUntil(this.stop$), shareReplay({ refCount: true, bufferSize: 1 }));
 
+    this.createClient = (type: string, clientConfig: Partial<ElasticsearchClientConfig> = {}) => {
+      const finalConfig = merge({}, config, clientConfig);
+      return this.createClusterClient(type, finalConfig, deps.http.getAuthHeaders);
+    };
+
     return {
       legacy: { config$: clients$.pipe(map(clients => clients.config)) },
-
-      adminClient,
-      dataClient,
       esNodesCompatibility$,
-
-      createClient: (type: string, clientConfig: Partial<ElasticsearchClientConfig> = {}) => {
-        const finalConfig = merge({}, config, clientConfig);
-        return this.createClusterClient(type, finalConfig, deps.http.getAuthHeaders);
-      },
+      adminClient: this.adminClient,
+      dataClient,
+      createClient: this.createClient,
     };
   }
 
-  public async start() {}
+  public async start() {
+    if (typeof this.adminClient === 'undefined' || typeof this.createClient === 'undefined') {
+      throw new Error('ElasticsearchService needs to be setup before calling start');
+    } else {
+      return {
+        client: this.adminClient,
```

Comment:
What if we reserve these property names for an elasticsearch-js compatible client? https://github.com/elastic/kibana/issues/35508
We can group them under `legacy` namespace, for example.

Reply:
I agree, this is a good time to prepare for that. It could be confusing if there's a "legacy" client, but we won't provide any alternative until 8.0, but I don't have a better suggestion. It would save a lot of refactoring and risk if we can prevent it, so I think it's worth the possible short term confusion for the long term benefits.

Reply:
> but we won't provide any alternative until 8.0, 

why not? I'd expect we add it in v7.8
<img src="https://user-images.githubusercontent.com/3198181/76605896-8f39ca00-6511-11ea-9bcc-de93b30c5ddb.png" width="300" height="400" />

Reply:
Ah, I somehow missed that!